Assets/Editor Toolbox/Editor/SceneView/ToolboxEditorSceneViewObjectSelector.cs

```diff
@@ -0,0 +1,344 @@
+                    continue;
+                }
+
+                GUIContent content = EditorGUIUtility.ObjectContent(gameObject, typeof(GameObject));
+
+                bool objectSelected = Selection.Contains(gameObject);
+
+                if (objectSelected)
+                {
+                    GUI.backgroundColor = selectionColor;
+                }
+
+                if (GUI.Button(rect, content, buttonStyle))
+                {
+                    GameObjectButtonPress(i);
+                }
+
+                GUI.backgroundColor = Color.white;
+
+                rect.y += buttonYSize + buttonYSpacing;
+            }
+        }
+
+        private void OnGUI_MouseMove()
+        {
+            Rect rect = new Rect(sizeXPadding, sizeYPadding, buttonSize.x, buttonSize.y);
+
+            for (int i = 0; i < gameObjects.Count; i++)
+            {
+                GameObject gameObject = gameObjects[i];
+
+                if (gameObject == null)
+                {
+                    //Can happen when something removes the gameobject during the window display.
+                    continue;
+                }
+
+                GUIContent content = EditorGUIUtility.ObjectContent(gameObject, typeof(GameObject));
+
+                GUI.Button(rect, content, buttonStyle);
+
+                if (rect.Contains(Event.current.mousePosition))
+                {
+                    HighlightedObject = gameObject;
+                }
+
+                rect.y += buttonYSize + buttonYSpacing;
+            }
+        }
+
+        private void GameObjectButtonPress(int id)
+        {
+            SelectObject(id, Event.current.control, Event.current.shift);
+
+            if (Event.current.control || Event.current.shift)
+            {
+                return;
+            }
+
+            Close();
+        }
+
+        private void UpdateShiftSelectionIDs(int id)
+        {
+            if(shiftLastID == -1)
+            {
+                shiftLastID = id;
+            }
+
+            if(shiftMinSelectionID == -1)
+            {
+                shiftMinSelectionID = id;
+            }
+
+            if (shiftMaxSelectionID == -1)
+            {
+                shiftMaxSelectionID = id;
+            }
+
+            if(id < shiftMinSelectionID)
+            {
+                shiftMinSelectionID = id;
+            }
+            else if(id >= shiftMaxSelectionID)
+            {
+                shiftMaxSelectionID = id;
+            }
+            else if(id > shiftMinSelectionID)
+            {
+                //ID is between min and max.
+                if(shiftLastID < id)
+                {
+                    shiftMaxSelectionID = id;
+                }
+                else
+                {
+                    shiftMinSelectionID = id;
+                }
+            }
+
+            shiftLastID = id;
+        }
+
+        private void SelectObject(int id, bool control, bool shift)
```

Comment:
Wouldn't it be better to rename control to append and shift to group or something similar?

Reply:
I think that control and shift is descriptive for what is happening here, since control is usually used for appending and shift for multi-selection over a span in many programs.

While calling control append would work, changing shift to grouping might be misleading, since control/appending also technically groups many objects together. I suggest keeping this version